components/causal_ts/src/observer.rs

```diff
@@ -97,8 +97,9 @@ impl<Ts: CausalTsProvider, Tk: RawTsTracker> QueryObserver for CausalObserver<Ts
                 ts = Some(self.causal_ts_provider.get_ts().map_err(|err| {
                     coprocessor::Error::Other(box_err!("Get causal timestamp error: {:?}", err))
                 })?);
+                // use prev ts as `resolved_ts` means the data with smaller or equal ts has already sink to cdc.
                 self.ts_tracker
-                    .track_ts(region_id, ts.unwrap())
+                    .track_ts(region_id, ts.unwrap().prev())
```

Comment:
Suggest to add test case for this change.

Reply:
I will add test in later PR with metric json, it's not easy to test this scenario.